src/Common/Services/NewEnvironmentUtilities.cs

```diff
@@ -16,5 +104,125 @@ public string GetDeploymentTargetValue(string abbreviatedDeploymentTarget)
+				},
+				new Filter()
+				{
+					Name = "Version",
+					ObjectName = "CustomerEnvironment",
+					ObjectAlias = "CustomerEnvironment_1",
+					Operator = Cmf.Foundation.Common.FieldOperator.IsNotEqualTo,
+					Value = customerEnvironment.Version,
+					LogicalOperator = Cmf.Foundation.Common.LogicalOperator.AND,
+					FilterType = Cmf.Foundation.BusinessObjects.QueryObject.Enums.FilterType.Normal,
+				},
+				new Filter()
+				{
+					Name = "UniversalState",
+					ObjectName = "CustomerEnvironment",
+					ObjectAlias = "CustomerEnvironment_1",
+					Operator = Cmf.Foundation.Common.FieldOperator.IsNotEqualTo,
+					Value = 4,
+					LogicalOperator = Cmf.Foundation.Common.LogicalOperator.AND,
+					FilterType = Cmf.Foundation.BusinessObjects.QueryObject.Enums.FilterType.Normal,
+				},
+				new Filter()
+				{
+					Name = "Status",
+					ObjectName = "CustomerEnvironment",
+					ObjectAlias = "CustomerEnvironment_1",
+					Operator = Cmf.Foundation.Common.FieldOperator.In,
+					Value = new int[] {
+						(int)DeploymentStatus.NotDeployed,
+						(int)DeploymentStatus.DeploymentFailed,
+						(int)DeploymentStatus.DeploymentPartiallySucceeded,
+						(int)DeploymentStatus.DeploymentSucceeded,
+						(int)DeploymentStatus.TerminationFailed
+					},
+					LogicalOperator = Cmf.Foundation.Common.LogicalOperator.AND,
+					FilterType = Cmf.Foundation.BusinessObjects.QueryObject.Enums.FilterType.Normal,
+				},
+				new Filter()
+				{
+					Name = "Name",
+					ObjectName = "CustomerEnvironment",
+					ObjectAlias = "CustomerEnvironment_1",
+					Operator = Cmf.Foundation.Common.FieldOperator.IsEqualTo,
+					Value = customerEnvironment.Name,
+					LogicalOperator = Cmf.Foundation.Common.LogicalOperator.Nothing,
+					FilterType = Cmf.Foundation.BusinessObjects.QueryObject.Enums.FilterType.Normal,
+				}
+			};
+			query.Query.Fields = new FieldCollection() {
+				new Field()
+				{
+					Alias = "Id",
+					ObjectName = "CustomerEnvironment",
+					ObjectAlias = "CustomerEnvironment_1",
+					IsUserAttribute = false,
+					Name = "Id",
+					Position = 0,
+					Sort = Cmf.Foundation.Common.FieldSort.NoSort
+				},
+				new Field()
+				{
+					Alias = "DefinitionId",
+					ObjectName = "CustomerEnvironment",
+					ObjectAlias = "CustomerEnvironment_1",
+					IsUserAttribute = false,
+					Name = "DefinitionId",
+					Position = 1,
+					Sort = Cmf.Foundation.Common.FieldSort.NoSort
+				},
+				new Field()
+				{
+					Alias = "Name",
+					ObjectName = "CustomerEnvironment",
+					ObjectAlias = "CustomerEnvironment_1",
+					IsUserAttribute = false,
+					Name = "Name",
+					Position = 2,
+					Sort = Cmf.Foundation.Common.FieldSort.NoSort
+				}
+			};
+			query.Query.Relations = new RelationCollection();
+
+			DataSet dataSet = await _customerPortalClient.ExecuteQuery(query);
+
+			var result = new CustomerEnvironmentCollection();
+			if (dataSet != null && dataSet.Tables.Count == 1 && dataSet.Tables[0].Rows.Count > 0)

```

Comment:
Couldn't this condition be simplified as you did in CustomerPortalClient line 235?

Reply:
the first part, yes it can!